File: Data Structures & Algorithms/subsets/submission-1.js
class Solution {
    ans = [];

    subsets(nums) {
        this.ans = [];
        this.solve(nums, 0, []);
        return this.ans;
    }

    solve(nums, idx, arr) {
        if (idx === nums.length) {
            this.ans.push([...arr]);
            return;
        }

        arr.push(nums[idx]);
        this.solve(nums, idx + 1, arr);

        arr.pop();

        this.solve(nums, idx + 1, arr);
    }
}

File: Data Structures & Algorithms/rotate-array/submission-1.java
class Solution {
    private void rot(int a[] , int i , int j){
        while(i <=j){
            int t = a[i];
            a[i] =a[j];
            a[j]=t;
            i++;
            j--;
        }
    }
    public void rotate(int[] nums, int k) {
        int n = nums.length ;
        k = k%n ;
                rot(nums,0,n-1);
        rot(nums,0,k-1);
        rot(nums,k,n-1);

    }
}